baseband/formats.py

```diff
@@ -0,0 +1,48 @@
+# Licensed under the GPLv3 - see LICENSE
+"""Routines to obtain information on baseband files."""
+import importlib
+
+
+FILE_FORMATS = {'dada': 'DADA',
+                'mark4': 'Mark4',
+                'mark5b': 'Mark5B',
+                'vdif': 'VDIF',
+                'gsb': 'GSB'}
+
+
+def file_info(name, fmt=None, **kwargs):
+    if fmt is None:
+        fmt = tuple(FILE_FORMATS.keys())
+
+    if isinstance(fmt, tuple):
+        for fmt_ in fmt:
+            info = file_info(name, fmt_, **kwargs)
+            if info:
+                break
+
+        return info
+
+    module = importlib.import_module('.' + fmt, package='baseband')
```

Comment:
I don't like `module` here, but can't think of something better (`baseband_format` seems excessive, and probably confusing).